contracts/src/v0.8/automation/dev/v2_3/AutomationRegistrar2_3.sol

```diff
@@ -177,10 +181,18 @@ contract AutomationRegistrar2_3 is TypeAndVersionInterface, ConfirmedOwner, IERC
    * @notice Allows external users to register upkeeps; assumes amount is approved for transfer by the contract
    * @param requestParams struct of all possible registration parameters
    */
-  function registerUpkeep(RegistrationParams calldata requestParams) external returns (uint256) {
-    if (!requestParams.billingToken.transferFrom(msg.sender, address(this), requestParams.amount)) {
-      revert TransferFailed(address(this));
+  function registerUpkeep(RegistrationParams memory requestParams) external payable returns (uint256) {
+    if (requestParams.billingToken == IERC20(WRAPPED_NATIVE_TOKEN_ADDRESS)) {
+      requestParams.amount = uint96(msg.value);
```

Comment:
I think we need two options here... "wrapped native" and "raw native". Ex what do we do if the users _already_ has wrapped native and genuinely wants to use their already wrapped native when registering? Here, we are kind of forcing them to use _raw_ native.

I think we could do a switch on `msg.value` ...  if `msg.value == 0` then try to do a `wrappedNative.transferFrom` otherwise call `wrappedNative.deposit()`

Reply:
also, we should `require(msg.value == 0)` for all other cases

Reply:
good call! updated to support the scenario you mentioned. 

Updated other places to use `msg.value == 0`, this place turns out to be dryer with msg.value != 0 :) 

```
    if (requestParams.billingToken == IERC20(WRAPPED_NATIVE_TOKEN) && msg.value != 0) {
      requestParams.amount = SafeCast.toUint96(msg.value);
      // wrap and send native payment
      WRAPPED_NATIVE_TOKEN.deposit{value: msg.value}();
    } else {
      // send ERC20 payment, including wrapped native token
      if (!requestParams.billingToken.transferFrom(msg.sender, address(this), requestParams.amount)) {
        revert TransferFailed(address(this));
      }
    }
```